lib/graphiti/util/transaction_hooks_recorder.rb

```diff
@@ -0,0 +1,46 @@
+module Graphiti
+  module Util
+    class TransactionHooksRecorder
+      class << self
+        def record
+          reset_hooks
+
+          begin
+            transaction_response = yield.tap do |r|
```

Comment:
I don't have huge objections to this, but if we're going with a pattern that requires a specific payload to be returned, let's add a comment at the top of this class noting that + basic usage.